federation/cmd/federation-apiserver/app/server.go

```diff
@@ -196,6 +197,8 @@ func Run(s *options.ServerRunOptions) error {
 		return err
 	}
 
+	routes.Version{}.Install(m.Mux, m.HandlerContainer)
```

Comment:
Am still undecided. The version package `base.go` is customized via ldflags. So this could be done by any instantiation of the genericapiserver.

The actual interpretation of the version values in the kubelet is done not in the generic discovery client itself, but in kubelet code (compare https://github.com/kubernetes/kubernetes/pull/32493/files).

The discovery client assumes a certain json schema. So at least this must be generic. I think the best thing is to inject a version.Info object into the genericapiserver. In the case of kube-apiserver this can be set via ldflags. But other instances of genericapiserver might choose different ways to return a proper version.Info.

@deads2k what's your opinion?